test/functional/outputs/azuremonitor/forward_to_azuremonitor_test.go

```diff
@@ -0,0 +1,194 @@
+		// this allow as emulate forwarding
+		hostname, err := getServerHost()
+		Expect(err).To(BeNil())
+		subDomain := strings.SplitN(hostname, ".", 2)
+		framework.Forwarder.Spec.Outputs = append(framework.Forwarder.Spec.Outputs,
+			logging.OutputSpec{
+				Name: logging.OutputTypeAzureMonitor,
+				Type: logging.OutputTypeAzureMonitor,
+				OutputTypeSpec: logging.OutputTypeSpec{
+					AzureMonitor: &logging.AzureMonitor{
+						LogType:    "myLogType",
+						CustomerId: fmt.Sprintf("%s-%s", azuremonitor.Mockoon, framework.Namespace),
+						Host:       fmt.Sprintf("apps.%s", subDomain[1]),
+					},
+				},
+				TLS: &logging.OutputTLSSpec{
+					InsecureSkipVerify: true,
+				},
+				Secret: &logging.OutputSecretSpec{
+					Name: azureSecret,
+				},
+			})
+
+		framework.Forwarder.Spec.Inputs = append(framework.Forwarder.Spec.Inputs,
+			logging.InputSpec{
+				Name:        inputName,
+				Application: &logging.Application{},
+			},
+		)
+		framework.Forwarder.Spec.Pipelines = append(framework.Forwarder.Spec.Pipelines,
+			logging.PipelineSpec{
+				InputRefs:  []string{inputName},
+				OutputRefs: []string{logging.OutputTypeAzureMonitor},
+				Name:       "pip-pip",
+			},
+		)
+
+		secret = runtime.NewSecret(framework.Namespace, azureSecret,
+			map[string][]byte{
+				constants.SharedKey: sharedKey,
+			},
+		)
+		framework.Secrets = append(framework.Secrets, secret)
+		Expect(framework.DeployWithVisitor(func(b *runtime.PodBuilder) error {
+			return mockServer.NewVisitor(b, framework)
+		})).To(BeNil())
+	})
+
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	It("should accept application logs", func() {
+		// Write application logs
+		timestamp := "2020-11-04T18:13:59.061892+00:00"
+		message := "This is my test message"
+		applicationLogLine := functional.NewCRIOLogMessage(timestamp, message, false)
+		Expect(framework.WriteMessagesToApplicationLog(applicationLogLine, 1)).To(BeNil())
+		time.Sleep(30 * time.Second)
+
+		//read log from collector container
+		collectorLog, err := framework.ReadCollectorLogs()
+		Expect(err).To(BeNil())
+		Expect(strings.Count(collectorLog, failedReason)).To(BeEquivalentTo(0))
+
+		//read log from mock server container and validate it
+		mockoonLogs := readLogFromMockoon(framework)
+		Expect(mockoonLogs).ToNot(BeNil())
+		Expect(len(mockoonLogs)).To(BeEquivalentTo(2))
+		Expect(mockoonLogs[0].Message).To(BeEquivalentTo(serverStarted))
+		Expect(mockoonLogs[1].RequestMethod).To(BeEquivalentTo(http.MethodPost))
+		Expect(mockoonLogs[1].Message).To(BeEquivalentTo(transactionRecorded))
+		Expect(mockoonLogs[1].Transaction.Request.Body).ToNot(BeEmpty())
+		var appLogs []types.ApplicationLog
+		err = types.ParseLogsFrom(utils.ToJsonLogs([]string{mockoonLogs[1].Transaction.Request.Body}), &appLogs, false)
+		Expect(err).To(BeNil())
+		Expect(len(appLogs)).To(BeEquivalentTo(1))
+		validateAppLog(appLogs[0], message, framework)
+	})
+
+	It("should accept batch application logs", func() {
```

Comment:
Is this different from the other test?  Are we not always sending logs as a "batch"?  I think likely we can either drop the other test or this one unless there is compelling reason to keep both